### PR TITLE
verbs in particle and recipe manifest + particle-by-verb and unnamed-connections resolution strategies

### PR DESCRIPTION
**Pull request by mmandlis**

### CODE REVIEW

File: runtime/particle-spec.js [hunk 1]
```diff
@@ -94,18 +95,20 @@ class ParticleSpec {
     return this.slots.get(slotName);
   }
 
+  get primaryVerb() { if (this.verbs.length > 0) return this.verbs[0]; }
```
Comment (dstockwell) [inline runtime/particle-spec.js]: Nit. Should be formatted over multiple lines.

File: runtime/strategies/search-tokens-to-particles.js [hunk 2]
```diff
@@ -17,9 +15,9 @@ module.exports = class SearchTokensToParticles extends Strategy {
     // TODO: Recipes. Views?
     this._byToken = {};
     for (let particle of arc.context.particles) {
-      let name = particle.name.toLowerCase();
-      this._byToken[name] = this._byToken[name] || [];
-      this._byToken[name].push(particle);
+      let verb = particle.primaryVerb;
```
Comment (dstockwell) [inline runtime/strategies/search-tokens-to-particles.js]: Should we support both particle and name here?

File: runtime/particle-spec.js (hunk 1 above)
Comment (mmandlis) [reply runtime/particle-spec.js]: done

File: runtime/strategies/search-tokens-to-particles.js (hunk 2 above)
Comment (mmandlis) [reply runtime/strategies/search-tokens-to-particles.js]: right, updated.
i also had to update the strategy itself to properly support the case with multiple particles corresponding to the same verb. and added a test case.

**Pull request merged**